spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -773,6 +801,11 @@ private File handleStringMessage(final String content, File originalFile, File t
 
 			@Override
 			protected void whileLocked() throws IOException {
+				if (newFileCallback != null && !fileToWriteTo.exists() && (FileExistsMode.APPEND.equals(fileExistsMode)
```

Comment:
I think this can be factored to some common code, but let it be for now.

I'll take a look into that during merge and polishing...

Let's bring it into the working shape first of all.

So, far I think you need to add a `FileWritingMessageHandler.this.` prefix into this property used in the inner classes.

Reply:
I have seen `this` codestyle errors, didn't know about `FileWritingMessageHandler.this` syntax, TIL :)

Of course,  some generalization could certainly be made, thanks!